packages/rest/src/resource.ts

```diff
@@ -191,21 +191,24 @@ export function $parentResource(context: DecoratorContext, entity: Type, parentT
   let currentType: Model | undefined = entity as Model;
   while (currentType) {
     const resourceKey = getResourceTypeKey(program, currentType);
-    const keyName = getKeyName(program, resourceKey.keyProperty);
-    if (keyNameSet.has(keyName)) {
-      reportDiagnostic(program, {
-        code: "duplicate-parent-key",
-        format: {
-          resourceName: (entity as Model).name,
-          parentName: currentType.name,
-          keyName,
-        },
-        target: resourceKey.keyProperty,
-      });
-      return;
+    if (resourceKey) {
```

Comment:
a bit confused on how this solve the problem(apart from removing the crash), wouldn't that basically just not make the parent child resource relation link and basically be the same as not using `@parentResource`?

Reply:
So that seems to still work in the playground if I add an resource interface for `WidgetPart` [link](https://cadlplayground.z22.web.core.windows.net/prs/966/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKQHNlcnZpY2VUaXRsZSgiV2lkZ2V0IFPGFSIpCm5hbWVzcGFjZSBEZW1vxxg7Cgp1c2luZyBDYWRsLkh0dHA7zBFSZXN0OwoKLy8gVGhpcyBpc3N1ZSBzb21laG93IHJlcXVpcmVzIHRocmVlIGZhY3RvcnMgdG8gYmUgaW4gcGxhY2U6xD0tIE1vcmUgdGhhbiBvbmUgcmVzb3VyY2UgdHlwZSBtdXN0IHNoYcUiZSBzYW1lIGBwYXJlbnRSxylgxkZUaGUgxhnUQ2JlIGJhc2VkIG9uIGEgdGVtcGxhdGVkxiJ3aGVyZSB5b3UgcGFzcyBhbm90aGVyIG1vZGVs5ADBb3VnaMtlcm9wZXJ0aeQA3cQ%2BdGhhdCBnZXRzxT7EVMYyxnloYXZlIGHINXnGLnJlZmXmAQ7kANUqc2Vjb25kKiBjaGls5gCT5QFXQ29tbWVudOQBcW91dCB05AFmxXJtYWvEfWhl5wF1Z28gYXdheQpA6AD5KCJwYXJ0cyIpCkDuASwo5gH6KQrmANzGDlBhcnQgewogIEBrZXkKIOQBRnRJZDogc3RyaW5nOwp9CsxabG9n31nJWUxvZ8xYbG9nz1fGLUJhc2XISDxUUOkBXz7FPeoBbzogyx3wAJd3xXLkAJrsAIIgaXPOX8YX0GTnAKHGRPUApNA4xTcvLyBOT1RFOuUCF%2BYBi3JlcHJvZHVjZXMgcmVnYXJkbGVzcyBvZuQCZOUCVeUBxmxpbsQxIHVuY%2BYB5OQCmXIgbm905wF%2FczrIcmFydFtdOwrGbusDactpd2hlbsZY6QJmZXhpc3RzIGJ1dCBnb2Vz5QIfxihp5QEUcmVtb3ZlZC7mAZbJbkxvZ8Rt5AFaZXJyb3LnAP1FxAzlAPJjb2RlOiBpbnQzMjsKICBtZXNzYWdl7QEwaW50ZXJm5ARLxlTnBE0gZXh0ZW5kcyDoAYvkBEDFCU9wZXJhdGlvbnPnAZ4sxnHnAZvkAcZAcm91dGUoImN1c3RvbUdldCIpIMkMKCnoAL%2FlAIDxAIHkAUNz33%2FMf8Qw6wCDfQo%3D) 

Don't really get how

Reply:
This part is only for checking if there's a conflicting key in the parent type.  The issue happens so rarely that I think it's OK in this case because we'd end up with parameter name conflicts elsewhere anyway.  The parent resource type is still stored at the top of the function, so all other functionality is retained.

Reply:
oh wait, this just does the validation, it doesn't actually set the property...

Reply:
so does that mean that you loose that duplicate parent key validation if you have this circular ref?
Should we not move this validation to `onValidate` maybe?

Reply:
Yep, it's just a validation check.

Reply:
yeah error disapear. I think we should add this validation to `onValidate` instead 
[playground repro](https://cadlplayground.z22.web.core.windows.net/prs/966/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKQHNlcnZpY2VUaXRsZSgiV2lkZ2V0IFPGFSIpCm5hbWVzcGFjZSBEZW1vxxg7Cgp1c2luZyBDYWRsLkh0dHA7zBFSZXN0xFtyZXNvdXJjZSgicGFydHMiKQpAcGFyZW50UsgZxm8pCm1vZGVsIMYOUGFydCB7CiAgQGtleQogIMQ8SWQ6IHN0cmluZzsKfc1abG9n31nJWUxvZ8xYac5UxipCYXNlyEU8VFByb3BlcnRpZXM%2BxTpwyRA6IMsd8ACUd8Vv5ACXzH8gaXPOX8YX0GT8AJ7QMsUxLy8gSWYgeW91IGNvbW1lbnQgb3V0IHRoZW7FFcR6dGhlIHZhbGlkYXRpb24gZXJyb3Ig5wFbczrIV2FydFtdOwogIOQBVMgWTG9nxBV9Cg%3D%3D)

Reply:
Checking during `onValidate` could certainly work, we can do that in another PR.  Allowing the decorator to push a post-compile validation job could be pretty sweet, I've wanted something like that a few times in the past.

Reply:
can you create an issue for it

Reply:
Filed #967.